tests/test_pipelines.py

```diff
@@ -398,6 +399,53 @@ def test_stable_diffusion_ddim(self):
         assert np.abs(image_slice.flatten() - expected_slice).max() < 1e-2
         assert np.abs(image_from_tuple_slice.flatten() - expected_slice).max() < 1e-2
 
+    def test_stable_diffusion_cycle(self):
+        device = "cpu"  # ensure determinism for the device-dependent torch.Generator
+        unet = self.dummy_cond_unet
+        scheduler = DDIMScheduler.from_config("CompVis/stable-diffusion-v1-4", subfolder="scheduler")
+        vae = self.dummy_vae
+        bert = self.dummy_text_encoder
+        tokenizer = CLIPTokenizer.from_pretrained("hf-internal-testing/tiny-random-clip")
+
+        # make sure here that pndm scheduler skips prk
+        sd_pipe = StableDiffusionCycleDiffusionPipeline(
+            unet=unet,
+            scheduler=scheduler,
+            vae=vae,
+            text_encoder=bert,
+            tokenizer=tokenizer,
+            safety_checker=self.dummy_safety_checker,
+            feature_extractor=self.dummy_extractor,
+        )
+        sd_pipe = sd_pipe.to(device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        source_prompt = "An astronaut riding a horse"
+        prompt = "An astronaut riding an elephant"
+        init_image = self.dummy_image.to(device)
+
+        generator = torch.Generator(device=device).manual_seed(0)
+        output = sd_pipe(
+            prompt=prompt,
+            source_prompt=source_prompt,
+            generator=generator,
+            num_inference_steps=2,
+            init_image=init_image,
+            eta=0.1,
+            strength=0.8,
+            guidance_scale=3,
+            source_guidance_scale=1,
+            output_type="np",
+        )
+        images = output.images
+
+        image_slice = images[0, -3:, -3:, -1]
```

Comment:
I'm currently getting `nan` values -> is this expected for small / dummy weights?

Reply:
I think it is possible when the image is random noises. This method infers the variance_noise that is likely to generate the input image, so I guess the variance_noise can be very large when the image is random noise.